### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by rodrigo-veloso**
Split clean data into 3 different functions

### CODE REVIEW

File: life_expectancy/cleaning.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import pandas as pd
+
+def load_data():
+    """
+    load data from file
+    """
+    data = pd.read_csv('life_expectancy/data/eu_life_expectancy_raw.tsv', sep='\t')
+    return data
+
+def save_data(data):
```
Comment (dgenio) [inline life_expectancy/cleaning.py:15]: Can you add input type hint?
```suggestion
def save_data(data: pd.DataFrame):
```

File: life_expectancy/cleaning.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
+"""
+Assignment 1
+"""
+import argparse
+import numpy as np
+import pandas as pd
+
+def load_data():
```
Comment (dgenio) [inline life_expectancy/cleaning.py:8]: Can you add a return type hint?
```suggestion
def load_data() -> pd.DataFrame:
```

File: life_expectancy/cleaning.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    load data from file
+    """
+    data = pd.read_csv('life_expectancy/data/eu_life_expectancy_raw.tsv', sep='\t')
+    return data
+
+def save_data(data):
+    """
+    save data to file
+    """
+    data.to_csv('life_expectancy/data/pt_life_expectancy.csv', index = False)
```
Comment (dgenio) [inline life_expectancy/cleaning.py:19]: Same comment as above - if not run from project root folder, it fails

File: life_expectancy/cleaning.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+def clean_data(data, country = 'PT'):
```
Comment (dgenio) [inline life_expectancy/cleaning.py:21]: Can you add more type hints?
```suggestion
def clean_data(data: pd.DataFrame, country: str = 'PT') -> pd.DataFrame:
```

File: life_expectancy/cleaning.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    """
+    Cleans data.
```
Comment (dgenio) [inline life_expectancy/cleaning.py:23]: Can you expand this docstring?
Explain the requirements of input variables ( columns that must be on data) and what the function does.

File: life_expectancy/cleaning.py [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    """
+    life_expectancy = data
```
Comment (dgenio) [inline life_expectancy/cleaning.py:25]: Nit: I would either just use data OR replace input variable name data with life_expectancy

File: life_expectancy/cleaning.py [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+    id_variables = ['unit','sex','age','region']
+
+    life_expectancy[id_variables] = (
+        life_expectancy['unit,sex,age,geo\\time'].str.split(',', expand = True)
+    )
+    life_expectancy = life_expectancy.drop(columns=['unit,sex,age,geo\\time'])
+    life_expectancy = life_expectancy[life_expectancy["region"] == country]
```
Comment (dgenio) [inline life_expectancy/cleaning.py:33]: If case is not correct, you might miss some filters
```suggestion
    life_expectancy = life_expectancy[life_expectancy["region"].str.upper() == country.str.upper()]
```

File: life_expectancy/cleaning.py [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+        id_vars = ['unit','sex','age','region'],
+        var_name = 'year',
+        value_name = 'value',
+    )
+    life_expectancy['year'] = life_expectancy['year'].str.replace(" ", "")
+    life_expectancy['year'] = life_expectancy['year'].astype(int)
+    life_expectancy['value'] = life_expectancy['value'].str.replace(" ", "")
+    life_expectancy['value'] = life_expectancy['value'].str.replace("e", "")
+    life_expectancy['value'] = life_expectancy['value'].replace(":",np.nan)
+    life_expectancy['value'] = life_expectancy['value'].astype(float)
```
Comment (dgenio) [inline life_expectancy/cleaning.py:48]: It is probably safer to do this with an expression...

File: life_expectancy/cleaning.py [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+    life_expectancy = life_expectancy.dropna()
+
+    return life_expectancy
+
+def main(country = 'PT'):
```
Comment (dgenio) [inline life_expectancy/cleaning.py:53]: Type hints
```suggestion
def main(country:str = 'PT') -> None:
```

**Review by dgenio — APPROVED** [review]

File: life_expectancy/cleaning.py [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+    life_expectancy = pd.melt(
+        life_expectancy,
+        value_vars = years,
+        id_vars = ['unit','sex','age','region'],
+        var_name = 'year',
+        value_name = 'value',
+    )
+    life_expectancy['year'] = life_expectancy['year'].str.replace(" ", "")
+    life_expectancy['year'] = life_expectancy['year'].astype(int)
+    life_expectancy['value'] = life_expectancy['value'].str.replace(" ", "")
```
Comment (taniacsilva) [inline life_expectancy/cleaning.py:45]:
```suggestion
     life_expectancy['value'] = life_expectancy['value'].str.replace(r"[a-zA-Z\s]+", "")
```

File: life_expectancy/cleaning.py [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+    life_expectancy['value'] = life_expectancy['value'].str.replace("e", "")
```
Comment (taniacsilva) [inline life_expectancy/cleaning.py:46]:
```suggestion
```

**Review by taniacsilva — APPROVED** [review]
+1

**Pull request merged**